cmdstanpy/cmdstan_args.py

```diff
@@ -192,7 +209,11 @@ def compose(self, idx: int, cmd: str) -> str:
             cmd = cmd + ' save_warmup=1'
         if self.thin is not None:
             cmd = '{} thin={}'.format(cmd, self.thin)
-        cmd = cmd + ' algorithm=hmc'
+        if self.fixed_param:
+            cmd = cmd + ' algorithm=fixed_param'
+            return cmd
```

Comment:
`return` nested this way is easy to miss given the pattern of the rest of the code in this method, but seems ok given that most it won't apply in the case of fixed params.